src/figuresClasses.ts

```diff
@@ -1,19 +1,98 @@
 export interface Figure {
-
+  shape: 'triangle' | 'rectangle' | 'circle';
+  getArea(): number;
+  color: 'green' | 'red' | 'blue';
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  side1: number;
+
+  side2: number;
+
+  side3: number;
 
+  constructor(color: 'red' | 'green' | 'blue',
+    side1: number, side2: number,
+    side3: number) {
+    this.color = color;
+    this.side1 = side1;
+    this.side2 = side2;
+    this.side3 = side3;
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again